WolvenKit.App/Helpers/TemplateFileHelper.cs

```diff
@@ -0,0 +1,417 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using WolvenKit.App.Interaction;
+using WolvenKit.App.Models.ProjectManagement.Project;
+using WolvenKit.Common.Services;
+using WolvenKit.Core.Interfaces;
+using WolvenKit.Interfaces.Extensions;
+using WolvenKit.RED4.Archive.CR2W;
+using WolvenKit.RED4.Types;
+using PhotoModePlayerEntityComponent = WolvenKit.RED4.Types.PhotoModePlayerEntityComponent;
+
+namespace WolvenKit.App.Helpers;
+
+public abstract class TemplateFileHelper
```

Comment:
I figured it doesn't need instantiation. I initially didn't want a service because a) dependency injection cycles, and b) this class isn't supposed to hold any kind of responsibility, it's just supposed to generate template files for whoever wants them. If you'd rather see it instantiated, I can possibly do that, though!

Reply:
1. Even if, it should be a `static` class, not an `abstract` one
2. I really don't like the fact that you introduce singleton patterns to other classes (which are properly registered in the DI container) just to make them accessible in a static context.

Reply:
The solution for that would be using Splat, but you told me not to... :/ How do we solve this?

Reply:
Why would you need splat?
Its only used in `AppViewModel`, where you could inject it via constructor injection.
Wouldn't also not need the singletons anymore, since you could add those to the TemplateFileHelper constructor.

Reply:
And then the cyclic dependency problem is back, and I need to add all those injectables to the constructors and/or hold them as instance variables, and... I just don't like it. Maybe I've grown up in the wrong parts of coding conventions. If there is a singleton and if it always exist, why do I have to do constructor injection to get it?

Reply:
> And then the cyclic dependency problem is back

That's a problem we all (me too) introduced. We stuffed way to much methods in classes (AppViewModel) where it doesn't belong and all code (including DI) depends on it.

Also I don't have a problem with using Singletons in general. But I learned at some point (could be wrong) that they should be avoided in C# / better to use DI. I see that they are way commoner in other languages like PHP.

Its less about if I can do something via singletons, but more that we agreed on using DI at some point.
Could be, that I'm just to strict about it.

> Maybe I've grown up in the wrong parts of coding conventions.

That could also apply to me.

Would be cool if some other people like @rfuzzo could chime in.
Maybe we should have a fundamental discussion about how to handle this stuff in the future.

Reply:
Solution we used at work for a legacy app was to introduce a ServiceProvider and inject _that_, so that we have one central singleton that spits out every other injectable when you poke it with a stick. It's also available via static class access. 

This would be a bigger refactoring in either case. Should we maybe postpone the solution of this problem and discuss it with the others?